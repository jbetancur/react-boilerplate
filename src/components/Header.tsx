import React from 'react';
import styled from 'styled-components';

const HeaderStyle = styled.header`
	display: flex;
	align-items: center;
	height: 58px;
	padding-left: 16px;
	padding-right: 16px;
	justify-content: space-between;
	background: ${({ theme }) => theme.palette.primary.main};
	color: ${({ theme }) => theme.palette.primary.contrastText};
`;

interface IHeaderProps {
	children: React.ReactNode;
}

const Header: React.FC<IHeaderProps> = ({ children }) => (
	<HeaderStyle>{children}</HeaderStyle>
);

export default Header;
