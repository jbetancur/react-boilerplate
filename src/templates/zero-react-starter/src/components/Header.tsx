import React from 'react';
import styled from 'styled-components';

const Title = styled.span`
	font-size: 24px;
	font-weight: 700;
	width: 100%;
`;

const Actions = styled.div`
	display: flex;
	justify-content: flex-end;
	width: 100%;

	a {
		color: ${({ theme }) => theme.palette.primary.contrastText};
		text-decoration: none;
	}

	> * {
		margin: 10px;
	}
`;

const HeaderStyle = styled.header`
	display: flex;
	align-items: center;
	height: 58px;
	padding-left: 16px;
	padding-right: 16px;
	background: ${({ theme }) => theme.palette.primary.main};
	color: ${({ theme }) => theme.palette.primary.contrastText};
`;

interface IHeaderProps {
	title?: string;
	children: React.ReactNode;
}

const Header: React.FC<IHeaderProps> = ({ title, children }) => (
	<HeaderStyle>
		<Title>{title}</Title>
		<Actions>{children}</Actions>
	</HeaderStyle>
);

export default Header;
