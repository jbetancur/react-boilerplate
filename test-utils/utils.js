import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { lightTheme, darkTheme } from '../src/themes';

const AllTheProviders = theme => ({ children }) => {
	const t = theme === 'light' ? lightTheme : darkTheme;

	return <ThemeProvider theme={t}>{children}</ThemeProvider>;
};

const renderWithTheme = (ui, options) => (theme = 'light') =>
	render(ui, { wrapper: AllTheProviders(theme), ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { renderWithTheme };
