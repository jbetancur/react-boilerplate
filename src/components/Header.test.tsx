import React from 'react';
import { renderWithTheme } from '../../test-utils/utils';
import Header from './Header';

test('should render correctly to default light theme', () => {
	const { container } = renderWithTheme(<Header>Noooooo!!!!!</Header>)('light');

	expect(container.firstChild).toMatchSnapshot();
});
