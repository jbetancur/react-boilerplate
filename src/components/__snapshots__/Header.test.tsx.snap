// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly to default light theme 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 58px;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background: #1976d2;
  color: #fff;
}

<header
  class="c0"
>
  Noooooo!!!!!
</header>
`;
